### PR TITLE
Jg us58 admin edit user

### PR DESCRIPTION
**Pull request by elguapogordo**
added test/function to admin user profile/password
no changes to database
209 examples, 0 failures

refactored:
split password controller off to its own
rerouted and relinked files
added testing for all three password input options
211 examples, 1 hershey/denver failure (at lunchtime, no less)

### CODE REVIEW

File: spec/features/admin/users/user_profile_spec.rb [hunk 1]
```diff
@@ -1,7 +1,7 @@
 require 'rails_helper'
 
 RSpec.describe "as an admin, when i visit a user profile page", type: :feature do
-  it "i see everything the user would see but no link to edit their profile" do
+  it "i have full access to that page including edit profile and password" do
```
Comment (elguapogordo) [inline spec/features/admin/users/user_profile_spec.rb:4]: undoing my previous user story code here...

File: app/controllers/admin/users_controller.rb [hunk 2]
```diff
@@ -8,4 +8,31 @@ def show
...
+      flash[:notice] = user.errors.full_messages.to_sentence
+      redirect_to "/admin/users/#{user.id}/edit"
+    end
+  end
+
+  def user_params
+    params.permit(:name, :address, :city, :state, :zip, :email, :password)
+  end
+
 end
```
Comment (steve-meyers) [inline app/controllers/admin/users_controller.rb:30]: Zach and I just did something similar and this logic won't guard against updating to a blank password. Also, we should make an admin password controller to handle the update.

**Review by steve-meyers — DISMISSED** [review]
Zach and I found a solution for password updates. Can you please update yours to match ours?

File: app/controllers/admin/users_controller.rb (hunk 2 above)
Comment (elguapogordo) [reply app/controllers/admin/users_controller.rb:30]: c/w

Comment (elguapogordo) [reply app/controllers/admin/users_controller.rb:30]: admin password controller c/w

**Comment by elguapogordo** [comment]
reworked to mirror zach's regular user update, now, as well

**Review by jrsewell400 — APPROVED** [review]
Changes look good!

File: app/controllers/admin/users_password_controller.rb [hunk 3]
```diff
@@ -0,0 +1,37 @@
+class Admin::UsersPasswordController < Admin::BaseController
```
Comment (zachholcomb) [inline app/controllers/admin/users_password_controller.rb:1]: nice work looks good

**Review by steve-meyers — APPROVED** [review]
Changes look good. Thanks!

**Pull request merged**